src/logging.h

```diff
@@ -203,6 +203,7 @@ static inline void LogPrintf_(const std::string& logging_function, const std::st
 #define LogPrintLevel_(category, level, ...) LogPrintf_(__func__, __FILE__, __LINE__, category, level, __VA_ARGS__)
 
 #define LogPrintf(...) LogPrintLevel_(BCLog::LogFlags::NONE, BCLog::Level::None, __VA_ARGS__)
+#define LogPrintfCategory(category, ...) LogPrintLevel_(category, BCLog::Level::None, __VA_ARGS__)
```

Comment:
Would be good to add a docstring for this. Is this meant to unconditionally log regardless of log level? If yes, it doesn't seem it would work. `BCLog::Level::None` is `1` but the check for unconditional logging is `>= Warning (3)`?

Reply:
Good catch; this is wrong for `LogPrintf` as well. Likely going to fix by reordering the `BCLog::Level` enums:
```diff
diff --git a/src/logging.cpp b/src/logging.cpp
index f1a86f0dce..3f316f0101 100644
--- a/src/logging.cpp
+++ b/src/logging.cpp
@@ -187,8 +187,6 @@ bool GetLogCategory(BCLog::LogFlags& flag, const std::string& str)
 std::string LogLevelToStr(BCLog::Level level)
 {
     switch (level) {
-    case BCLog::Level::None:
-        return "none";
     case BCLog::Level::Debug:
         return "debug";
     case BCLog::Level::Info:
@@ -197,6 +195,8 @@ std::string LogLevelToStr(BCLog::Level level)
         return "warning";
     case BCLog::Level::Error:
         return "error";
+    case BCLog::Level::None:
+        return "";
     }
     assert(false);
 }
```
```diff
diff --git a/src/logging.h b/src/logging.h
index 8a896b6b33..f8e55b88b2 100644
--- a/src/logging.h
+++ b/src/logging.h
@@ -69,10 +69,10 @@ namespace BCLog {
     };
     enum class Level {
         Debug = 0,
-        None = 1,
-        Info = 2,
-        Warning = 3,
-        Error = 4,
+        Info,
+        Warning,
+        Error,
+        None,
     };
```

Reply:
>  it doesn't seem it would work. BCLog::Level::None is 1 but the check for unconditional logging is >= Warning (3)?

Ah, `LogAcceptCategory()` with the logic you mention is called by neither `LogPrintf` nor `LogPrintfCategory`, so it seems fine as-is. Will add a docstring.

Reply:
> Would be good to add a docstring for this.

Thanks for the suggestion; done.